src/components/navigation/header/config/HeaderNavigationConfig.tsx

```diff
@@ -54,6 +59,7 @@ export class HeaderNavigationConfig extends BaseConfig<HeaderNavigationConfigIns
   homeUrl: string;
   menuHeight: string;
   user: User | null;
+  userType: UserType;
```

Comment:
is there a reason you need to specifically pass this and not use `user.type`? I see that it should be part of the user prop

Reply:
We need it at this level, because we are doing some visibility checks when we are constructing link. `user` could be `null`, but we still need `userType` (it will be `guest` in case `user` is `null`) to determine link visibility. 
It is just for internal use, client doesn't have to pass `userType` as prop to navigation.